### DISCUSSION TITLE
Does the `decode` and `encode`  of Query string  can be deprecated?

### DISCUSSION THREAD

**Discussion by iam-frankqiu** [nodejs/help#4828] [Q&A]
Because the methods of `decode` and `encode` just an alias of `parse` and `stringify`. So I think it should be deprecated.

**Comment by Trott** [comment]
Deprecation will cause hardship for end users and won't result in any significant benefit for the maintainers or the end users. I'm not opposed if there's a good reason to do this, but we don't need more deprecations or fewer aliases or anything like that.

**Reply by iam-frankqiu** [reply]
> Deprecation will cause hardship for end users and won't result in any significant benefit for the maintainers or the end users. I'm not opposed if there's a good reason to do this, but we don't need more deprecations or fewer aliases or anything like that.

I think we need a more simper API for end users. A simper API is a good thing for end users and maintainers, code reviewers. We can deprecate this API step by step, not one step.  A simple API and high-performance Node.js  with Promise, not Callback is worth us to use and maintain. That's my opinion. Thank you.
